### ISSUE TITLE
"warning: argument unused during compilation" when compiling assembly files

### ISSUE THREAD

**Issue by jzck** [ziglang/zig#2741]
Hello, as of recently assembly files seems to be misbehaving, I've reproduced a minimal viable example:

```
// main.zig
pub fn main() noreturn {
    while (true) {}
}

// test.s
.global __test
.type __test, @function

__test:
      hlt

//build.zig
const Builder = @import("std").build.Builder;

pub fn build(b: *Builder) void {
    const kernel = b.addExecutable("exe", "main.zig");
    kernel.setOutputDir("build");
    kernel.addAssemblyFile("test.s");

    b.default_step.dependOn(&kernel.step);
}
```

which gives me the output
```
~/zig_test$ zig build
zig: warning: argument unused during compilation: '-MD' [-Wunused-command-line-argument]
zig: warning: argument unused during compilation: '-MV' [-Wunused-command-line-argument]
zig: warning: argument unused during compilation: '-MF /home/jack/zig_test/zig-cache/tmp/6EiTG286E8Bm-test.o.d' [-Wunused-command-line-argument]
zig: warning: argument unused during compilation: '-nostdinc' [-Wunused-command-line-argument]
zig: warning: argument unused during compilation: '-fno-spell-checking' [-Wunused-command-line-argument]
zig: warning: argument unused during compilation: '-isystem /snap/zig/640/lib/zig/include' [-Wunused-command-line-argument]
zig: warning: argument unused during compilation: '-D _DEBUG' [-Wunused-command-line-argument]
zig: warning: argument unused during compilation: '-fno-stack-protector' [-Wunused-command-line-argument]
zig: warning: argument unused during compilation: '-fno-omit-frame-pointer' [-Wunused-command-line-argument]
```

These are just warnings, but I'm suspecting it's causing some of the bugs I'm experiencing. I'll look into how zig handles assembly if I have the time.

PS: is there a roadmap concerning assembly in zig for the future somewhere? I'm interested to see what the plan is

**Comment by andrewrk** [comment]
Those unused warnings are the result of zig not distinguishing between using `zig cc` to compile C code or assembly code. They're harmless (although I'll leave this issue open to fix them). I think your issue is something else.

> PS: is there a roadmap concerning assembly in zig for the future somewhere? I'm interested to see what the plan is

There are a few open issues, such as #2081 and #215.

**Comment by andrewrk** [comment]
Planned solution for this is #3508.

**Comment by Vexu** [comment]
Fixed by #3508
